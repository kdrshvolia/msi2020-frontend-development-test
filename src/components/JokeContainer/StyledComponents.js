import styled, { css } from 'styled-components';

const JokeInfo = styled.div`
  display: flex;
  justify-content: space-between;
`;

const FavoritedIcon = styled.div`
  display: flex;
  & > svg {
    margin-left: auto;
  }
  padding: 10px 0;
`;

const Category = styled.div`
  padding: 6px 20px;
  box-sizing: border-box;
  width: 107px;
  background: #ffffff;
  border-radius: 6px;
  font-weight: 500;
  font-size: 10px;
  line-height: 14px;
  letter-spacing: 2px;
  text-transform: uppercase;
  text-align: center;
`;

const Joke = styled.div`
  display: flex;
`;

const JokeWrapper = styled.div`
  display: flex;
  padding: 20px;
  flex-direction: column;
  width: 100%;
  max-width: ${(props) => (props.isInFavoritedList ? '400px' : '680px')};
  background: ${(props) => (props.isInFavoritedList ? '#ffffff' : '#f8f8f8')};
  border-radius: 20px;
  box-sizing: border-box;
  padding: ${(props) => (props.isInFavoritedList ? '20px' : '45px')};
  font-weight: normal;
  align-items: top;
  margin: 10px 0;
  box-shadow: ${(props) => (props.isInFavoritedList ? '0px 4px 6px rgba(0, 0, 0, 0.1)' : 'none')};
`;

const JokeText = styled.p`
  font-weight: normal;
  font-size: 18px;
  line-height: 26px;
`;

const UpdateInfo = styled.div`
  font-size: 10px;
  line-height: 14px;
  width: 128px;
  color: #ababab;
`;

const Link = styled.div`
  color: #8ea7ff;
  & > a {
    color: #8ea7ff;
  }
`;

const Logo = styled.div`
  display: flex;
  align-items: flex-start;
  padding: 0 20px 0 0;
`;

export { JokeWrapper, JokeText, UpdateInfo, Category, JokeInfo, Link, Logo, Joke, FavoritedIcon };
