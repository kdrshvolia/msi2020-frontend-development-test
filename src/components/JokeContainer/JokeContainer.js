import React from 'react';
import PropTypes from 'prop-types';
import { calculateTimeDifference } from '../../util';
import { JokeText, JokeWrapper, UpdateInfo } from './StyledComponents';

const JokeContainer = ({ joke }) => {
  return (
    <JokeWrapper>
      <div>
        <div className="logo" />
        <a href={joke.url}>{joke.id}</a>
        <JokeText className="joke-text">{joke.value}</JokeText>
        <div className="category">{joke.categories} </div>
        <UpdateInfo className="last-update">
          Last update: {calculateTimeDifference(Date.parse(joke.updated_at), new Date())} hours ago.
        </UpdateInfo>
      </div>
    </JokeWrapper>
  );
};

export default JokeContainer;
