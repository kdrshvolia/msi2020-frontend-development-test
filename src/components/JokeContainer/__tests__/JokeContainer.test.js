import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import JokeContainer from '../JokeContainer';

const joke = {
  id: '1',
  url: 'url',
  value: 'some value',
  updated_at: 'date',
  icon_url: 'url',
  categories: ['category'],
};
const isClicked = jest.fn();

describe('JokeContainer', () => {
  it('toggles favorite on click', () => {
    const wrapper = mount(<JokeContainer toggleFav={isClicked} joke={joke} />);
    const icon = wrapper.find('svg');
    icon.simulate('click', joke.id);
    expect(isClicked).toBeCalledWith(joke.id);
  });
  it('Renders correctly', () => {
    const rendered = renderer.create(
      <JokeContainer toggleFav={isClicked} joke={joke} isFavorited isInFavoritedList />,
    );
    expect(rendered).toMatchSnapshot();
  });
});
