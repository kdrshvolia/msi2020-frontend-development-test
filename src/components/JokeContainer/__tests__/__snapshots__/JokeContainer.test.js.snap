// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JokeContainer Renders correctly 1`] = `
<div
  className="sc-AxhUy eFfkTv"
>
  <div
    className="sc-AxirZ dSMUgx"
  >
    <svg
      height="17"
      onClick={[Function]}
      viewBox="0 0 20 17"
      width="20"
    >
      <path
        d="M18.4134 1.66367C17.3781 0.590857 15.9575 0 14.413 0C13.2585 0 12.2012 0.348712 11.2704 1.03637C10.8008 1.38348 10.3752 1.80814 10 2.3038C9.62494 1.80829 9.19922 1.38348 8.7294 1.03637C7.79877 0.348712 6.74149 0 5.58701 0C4.04251 0 2.62177 0.590857 1.58646 1.66367C0.563507 2.72395 0 4.17244 0 5.74252C0 7.35852 0.630341 8.83778 1.98364 10.3979C3.19427 11.7935 4.93423 13.2102 6.94916 14.8507C7.63718 15.411 8.41705 16.046 9.22684 16.7224C9.44077 16.9015 9.71527 17 10 17C10.2846 17 10.5592 16.9015 10.7729 16.7227C11.5826 16.0461 12.363 15.4108 13.0513 14.8503C15.0659 13.2101 16.8059 11.7935 18.0165 10.3978C19.3698 8.83778 20 7.35852 20 5.74238C20 4.17244 19.4365 2.72395 18.4134 1.66367Z"
        fill="#ff6767"
      />
    </svg>
  </div>
  <div
    className="sc-AxhCb ZaNiK"
  >
    <div
      className="sc-AxmLO hIXVty"
    >
      <img
        alt=""
        src="static-logo-fav.svg"
      />
    </div>
    <div>
      <div
        className="sc-Axmtr WCurB"
      >
        ID: 
        <a
          href="url"
          rel="noopener noreferrer"
          target="_blank"
        >
          1
        </a>
         
        <img
          alt=""
          src="link.svg"
        />
      </div>
      <p
        className="sc-AxgMl gQnCLI joke-text"
      >
        some value
      </p>
    </div>
  </div>
  <div
    className="sc-AxjAm bveoqz"
  >
    <div
      className="sc-AxheI gXKdnG last-update"
    >
      Last update: 
      NaN
       hours ago.
    </div>
    <div
      className="sc-AxiKw fUKOnE category"
    >
      category
       
    </div>
  </div>
</div>
`;
